addons/web_editor/tools.py

```diff
@@ -0,0 +1,132 @@
+        (or False in case of invalid URL).
+    '''
+    if not video_url:
+        return False
+
+    # To detect if we have a valid URL or not
+    valid_url_regex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+    # Regex for few of the widely used video hosting services
+    youtube_regex = r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})(?:\S+)?$'
+    vimeo_regex = r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*'
+    vine_regex = r'\/\/vine.co\/v\/(.[a-zA-Z0-9]*)'
+    dailymotion_regex = r'(?:.+dailymotion.com\/(video|hub|embed\/video|embed)|dai\.ly)\/([^_?]+)[^#]*(#video=([^_&]+))?'
+    instagram_regex = r'(?:(.*)instagram.com|instagr\.am)\/p\/(.[a-zA-Z0-9-_\.]*)'
+    youku_regex = r'(.*).youku\.com\/(v_show\/id_|embed\/)(.+)'
+
+    if re.search(valid_url_regex, video_url):
+        youtube_match = re.search(youtube_regex, video_url)
+        vimeo_match = re.search(vimeo_regex, video_url)
+        vine_match = re.search(vine_regex, video_url)
+        dailymotion_match = re.search(dailymotion_regex, video_url)
+        instagram_match = re.search(instagram_regex, video_url)
+        youku_match = re.search(youku_regex, video_url)
+
+        if youtube_match and len(youtube_match.groups()[1]) == 11:
+            return ('youtube', youtube_match.groups()[1], youtube_match)
+        elif vimeo_match:
+            return ('vimeo', vimeo_match.groups()[2], vimeo_match)
+        elif vine_match:
+            return ('vine', vine_match.groups()[0], vine_match)
+        elif dailymotion_match:
+            return ('dailymotion', dailymotion_match.groups()[1], dailymotion_match)
+        elif instagram_match:
+            return ('instagram', instagram_match.groups()[1], instagram_match)
+        elif youku_match:
+            youku_link = youku_match.groups()[2]
+            if '.html?' in youku_link:
+                youku_link = youku_link.split('.html?')[0]
+            return ('youku', youku_link, youku_match)
+    return False
+
+def get_video_url_data(video_url, **kwargs):
+    source = get_video_platform_and_id(video_url)
+    if not source:
+        return {'error': True, 'message': _('The provided url is invalid')}
+
+    autoplay = kwargs.get('autoplay') and '?autoplay=1&mute=1' or '?autoplay=0'
+    controls = kwargs.get('hide_controls') and '&controls=0' or ''
+    loop = kwargs.get('loop') and '&loop=1' or ''
+
+    # We directly use the provided URL as it is
+    embed_url = video_url
+    platform = source[0]
+    platform_id = source[1]
+    platform_match = source[2]
+    if platform == 'youtube' and len(platform_id) == 11:
+        fullscreen = kwargs.get('hide_fullscreen') and '&fs=0' or ''
+        youtube_loop = loop and loop + '&playlist=%s' % (platform_id) or ''
+        logo = kwargs.get('hide_yt_logo') and '&modestbranding=1' or ''
```

Comment:
> shouldn't we use color=white at the same time to really remove the youtube branding ?? (suggestion)
> (ps: it disabled at the same time the modestbranding, but you can keep it, explicit is better than implicit)
> 
> * add showinfo=0

showinfo has been deprecated in the player parameters. 
see: https://developers.google.com/youtube/player_parameters#showinfo

Reply:
not really deprectaed, but since it will only work for ActionScript 3, yes we can probably ignore it.
thx

Reply:
not sure, let me 2 min to make a test...

I don't see where is it deprecated.
There are just a warning for ActionScript3 potential bug if you use showinfo=1 with playlist.
But is doesn't means that it will not works when you publish one video (the most frequent usecase in Odoo)

Reply:
Okey, If you want me to add `showinfo=0`, I will. I was just following the official youtube IFrame Player API mentioned in the above link which says 

> The showinfo parameter, which indicates whether the player should display information like the video title and uploader before the video starts playing, is also being deprecated. Following the change, the channel avatar and video title will always display before playback begins, when playback is paused, and when playback ends. The avatar being displayed is new behavior that will be consistent across all embedded players.

and this deprecation is effective since September 25, 2018, according to the documentation

Reply:
Ok sorry, in this case, just ignore it ;) my bad !

I didn't see it :/ Seems like we don't have the same content for the link ...

Probably the french page is not updated ;)
![image](https://user-images.githubusercontent.com/4848797/87029069-e6aa2700-c1df-11ea-9b49-34a186898f97.png)